### PR TITLE
fix: correct object return types for arrow functions

### PR DESCRIPTION
**Pull request by Mause**

### CODE REVIEW

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp [hunk 1]
```diff
@@ -21,6 +21,30 @@
...
+class arrow_table : public py::object {
+public:
+	arrow_table(const py::object &o) : py::object(o, borrowed_t {}) {
+	}
+	using py::object::object;
+
+	static bool check_(const py::handle &object) {
+		return !py::none().is(object);
+	}
+};
```
Comment (pdet) [inline tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:45]: The class names and methods don't comply with the standard layout. In particular, class method nomenclature.


For example:
``` cpp

class MyClass {
public:
	MyClass();

	int my_public_variable;

public:
	void MyFunction();

private:
	void MyPrivateFunction();

private:
	void my_private_variable;
};
```

File: tools/pythonpkg/src/pyrelation.cpp [hunk 2]
```diff
@@ -94,8 +94,6 @@ void DuckDBPyRelation::Initialize(py::handle &m) {
 	         py::arg("aggregation_columns"))
 	    .def("describe", &DuckDBPyRelation::Describe,
 	         "Gives basic statistics (e.g., min,max) and if null exists for each column of the relation.")
-	    .def("union", &DuckDBPyRelation::Union,
```
Comment (pdet) [inline tools/pythonpkg/src/pyrelation.cpp:97]: why is this removed?

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp (hunk 1 above)
Comment (Mause) [reply tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:45]: Oh cool, happy to fix that, is that documented anywhere?

File: tools/pythonpkg/src/pyrelation.cpp (hunk 2 above)
Comment (Mause) [reply tools/pythonpkg/src/pyrelation.cpp:97]: As mentioned in the commit, it's a duplicate

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp (hunk 1 above)
Comment (pdet) [reply tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:45]: Yes, in the contributions.md at the root folder.

It should also be enforced by the Tidy-Check CI.

**Comment by pdet** [comment]
Besides some nitpicks lgtm ;-), thanks for the fixes!

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp (hunk 1 above)
Comment (Mause) [reply tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp]: @pdet I've made the changes that appear to be required per that style, lmk if those are sufficient

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp [hunk 3]
```diff
@@ -46,3 +72,16 @@ class PythonTableArrowArrayStreamFactory {
...
+
+namespace pybind11 {
+namespace detail {
+template <>
+struct handle_type_name<duckdb::pyarrow::RecordBatchReader> {
+	static constexpr auto name = _("pyarrow.lib.RecordBatchReader");
+};
+template <>
+struct handle_type_name<duckdb::pyarrow::Table> {
+	static constexpr auto name = _("pyarrow.lib.Table");
```
Comment (pdet) [inline tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:84]: Constexpr should be uppercase

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp [hunk 4]
```diff
@@ -21,6 +21,32 @@
...
+	}
+};
+class Table : public py::object {
+public:
+	Table(const py::object &o) : py::object(o, borrowed_t {}) {
+	}
+	using py::object::object;
+
+public:
+	static bool check_(const py::handle &object) {
```
Comment (pdet) [inline tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:44]: methods shouldn't finish with _

**Comment by pdet** [comment]
It's clear that the tidy-check CI is not running on the python code, since the dataframe class was already merged, I'm having a look on why that's the case, probably got lost on some prior CI work.

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp (hunk 4 above)
Comment (Mause) [reply tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:44]: This name is from pybind11, I can't change it

File: tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp (hunk 3 above)
Comment (Mause) [reply tools/pythonpkg/src/include/duckdb_python/arrow_array_stream.hpp:84]: This name is from pybind11, I can't change it

**Comment by pdet** [comment]
Alright, lgtm, if you have no comments @Mytherin , this is good to go imo

**Comment by Mytherin** [comment]
Thanks!
